### DISCUSSION TITLE
docker image update recognition `release-x.x`

### DISCUSSION THREAD

**Discussion by jokay** [renovatebot/renovate#9502] [Request Help] [answered]
Given [this](https://hub.docker.com/r/xibosignage/xibo-cms/tags) container on hub.docker.com which uses tags in the format of `:release-x.x`.

Should the recognition for updates work for this format as well or is not supposed to work?

**Comment by viceice** [comment, marked as answer]
You need a custom regex version scheme

**Reply by viceice** [reply]
See this sample

https://github.com/visualon/docker-images/blob/b8f34e1f87829f77e6c588817fa3f5802c99b1ff/.github/renovate.json#L40

**Reply by jokay** [reply]
nice, didn't know how much customizable renovate is, thx a lot :+1:

**Reply by jokay** [reply]
@viceice ok, I tried several different ways but can't get it to work, this is what I have currently but still doesn't work 🧐 

It should update from `xibosignage/xibo-cms:release-2.3.9` -> `xibosignage/xibo-cms:release-2.3.10`

```json
{
    "packageNames": [
        "xibosignage/xibo-cms"
    ],
    "extractVersion": "^release-(?<version>.*)$"
}
```

[Here](https://docs.renovatebot.com/configuration-options/#extractversion) they state "Alternatively, to strip a release- prefix" but it still doesn't work.

**Reply by viceice** [reply]
Please provide a small reproduction repo

**Reply by jokay** [reply]
It runs on a self-hosted GitLab instance, but this is mainly it:

#### docker-compose.yml

```yml
services:
    app:
      image: xibosignage/xibo-cms:release-2.3.9
```

#### renovate.json

```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:base",
        "docker:enableMajor"
    ],
    "prHourlyLimit": 0,
    "prConcurrentLimit": 0,
    "packageRules": [
        {
            "packageNames": [
                "xibosignage/xibo-cms"
            ],
            "extractVersion": "^release-(?<version>.*)$"
        }
    ]
}
```

**Reply by viceice** [reply]
extractVersion doesn't work here, you need a regex manager

**Reply by jokay** [reply]
Are there any (working) examples for existing use cases, like strip a "release-" prefix as stated in the documentation, I mentionned?

**Reply by viceice** [reply]
Your compose file needs the full version string  updated, so you can't use extract.

That's why I told you to use a custom regex versioning like the one sample i linked.

**Reply by jokay** [reply]
Will try that, thx.

**Reply by viceice** [reply]
`extractVersion` is for use case where the datasource is delivering long version tags and you source file needs some short version.

Like in

https://github.com/visualon/docker-images/blob/b8f34e1f87829f77e6c588817fa3f5802c99b1ff/.github/renovate.json#L68  


https://github.com/visualon/docker-images/blob/b8f34e1f87829f77e6c588817fa3f5802c99b1ff/linux/github-runner/Dockerfile#L35

**Reply by jokay** [reply]
I created a repro repo on GitHub.

It seems to work on GitHub, no idea where the "v" comes from in the PR title.

But the exact same `renovate.json` does not work on a self-hosted GitLab instance (`renovate/renovate:24.113.0`).

**Reply by jokay** [reply]
The only MR which gets created is this one:

![pr](https://user-images.githubusercontent.com/18613935/114499798-1bdc1d80-9c27-11eb-9aab-2996ca0ff213.png)

**Reply by jokay** [reply]
I found the cause. If adding the registry, e.g. `docker.io` (`ghcr.io`) it won't work.

**Reply by rarkins** [reply]
There's no difference in this type of functionality between GitHub and Gitlab or app and self hosted. The difference must be coming from your config somehow.

**Reply by jokay** [reply]
@rarkins it seems to be a problem of registry and regex combination, see my [comment](https://github.com/renovatebot/renovate/discussions/9502#discussioncomment-603178) above.

**Reply by jokay** [reply]
```sh
DEBUG: Dependency docker.io/gitlab/gitlab-runner has unsupported value alpine-v13.9.0
DEBUG: getLabels(https://index.docker.io, gitlab/gitlab-ce, latest)
DEBUG: getLabels(https://index.docker.io, xibosignage/xibo-cms, latest)
DEBUG: getLabels(https://index.docker.io, gitlab/gitlab-runner, latest)
DEBUG: getManifestResponse(https://index.docker.io, gitlab/gitlab-runner, latest)
DEBUG: getLabels(https://index.docker.io, xibosignage/xibo-xmr, latest)
DEBUG: getManifestResponse(https://index.docker.io, gitlab/gitlab-runner, sha256:2d36789489d179d1c472d37d53e8920c0593b8da117961fb3d38d2fe86c3d40a)
```

Having `docker.io` as  part of the image results in:

```sh
DEBUG: Dependency docker.io/gitlab/gitlab-runner has unsupported value alpine-v13.9.0`
```

Without `docker.io` as part of the image it works:

```sh
DEBUG: getLabels(https://index.docker.io, gitlab/gitlab-runner, latest)
DEBUG: getManifestResponse(https://index.docker.io, gitlab/gitlab-runner, latest)
DEBUG: getManifestResponse(https://index.docker.io, gitlab/gitlab-runner, sha256:2d36789489d179d1c472d37d53e8920c0593b8da117961fb3d38d2fe86c3d40a)
```

**Reply by rarkins** [reply]
You need to adjust the packageNames values in your packageRules

**Reply by jokay** [reply]
Good point 😉 

Works perfectly now, thank you very much.
